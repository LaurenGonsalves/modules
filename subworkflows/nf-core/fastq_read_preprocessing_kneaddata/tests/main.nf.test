nextflow_workflow {

    name "Test Subworkflow: FASTQ_READ_PREPROCESSING_KNEADDATA"
    script "../main.nf"
    workflow "FASTQ_READ_PREPROCESSING_KNEADDATA"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_read_preprocessing_kneaddata"
    tag "fastq_read_preprocessing_kneaddata_default"


    test(" fasta.gz + 'human_genome' ") {

        when {
            params {
                max_cpus = 1
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ],
                    [
                        [ id: 'test2' ],
                        [
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test2_1.fastq.gz", checkIfExists: true),
                            file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test2_2.fastq.gz", checkIfExists: true)
                        ]
                    ]
                )
                input[1] = null
                input[2] = 'human_genome'
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
