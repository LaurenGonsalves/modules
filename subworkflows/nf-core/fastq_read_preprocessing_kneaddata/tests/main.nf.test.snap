{
    " fasta.gz + 'human_genome' ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test2_kneaddata_paired_2.fastq.gz:md5,a544916a192f082f804b21d8c2d2c186"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,de930236265c9d98bfac038bc471d3f0"
                        ]
                    ]
                ],
                "1": [
                    "all_samples_kneaddata_read_counts.tsv:md5,0cbc1c285d40c25876ce0675fe36f34b"
                ],
                "2": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8",
                    "versions.yml:md5,c97c9af1b0da5c6c85c297ad365653cc"
                ],
                "preprocessed_fastq_gz": [
                    [
                        {
                            "id": "test2"
                        },
                        [
                            "test2_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test2_kneaddata_paired_2.fastq.gz:md5,a544916a192f082f804b21d8c2d2c186"
                        ]
                    ],
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,de930236265c9d98bfac038bc471d3f0"
                        ]
                    ]
                ],
                "read_counts_tsv": [
                    "all_samples_kneaddata_read_counts.tsv:md5,0cbc1c285d40c25876ce0675fe36f34b"
                ],
                "versions": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8",
                    "versions.yml:md5,c97c9af1b0da5c6c85c297ad365653cc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-25T22:21:23.676497909"
    }
}